internal/core/services/worker/worker.go

```diff
@@ -0,0 +1,122 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/services/operation_manager"
+	"github.com/topfreegames/maestro/internal/core/services/operations_registry"
+	"go.uber.org/zap"
+)
+
+// Worker is the service responsible for implemeting the worker
+// responsabilities.
+type Worker struct {
+	opManager *operation_manager.OperationManager
+	// TODO(gabrielcorado): check if we this is the right place to have all
+	// executors.
+	executors map[string]operations.Executor
+
+	logger *zap.Logger
+}
+
+type WorkerOptions struct {
+	OperationStorage   ports.OperationStorage
+	OperationFlow      ports.OperationFlow
+	Executors          []operations.Executor
+	OperationsRegistry *operations_registry.Registry
+}
+
+func NewWorker(opts *WorkerOptions) *Worker {
+	opManager := operation_manager.New(opts.OperationFlow, opts.OperationStorage)
+	if opts.OperationsRegistry != nil {
+		opManager = operation_manager.NewWithRegistry(opts.OperationFlow, opts.OperationStorage, *opts.OperationsRegistry)
+	}
+
+	executors := make(map[string]operations.Executor)
+	for _, executor := range opts.Executors {
+		// TODO(gabrielcorado): are we going to receive the executor
+		// initialized?
+		executors[executor.Name()] = executor
+	}
+
+	return &Worker{
+		opManager: opManager,
+		executors: executors,
+		logger:    zap.L().With(zap.String("service", "worker")),
+	}
+}
+
+// ShouldExecuteOperaiton is responsible for starting a loop that will
+// constantly look to execute operations, and this loop can be canceled using
+// the provided context. NOTE: It is a blocking function.
+func (w *Worker) SchedulerOperationsExecutionLoop(ctx context.Context, schedulerName string) error {
+	for {
+		op, def, err := w.opManager.NextSchedulerOperation(ctx, schedulerName)
+		if err != nil {
+			if errors.Is(err, context.Canceled) {
+				return nil
+			}
+
+			return fmt.Errorf("failed to get next operation: %w", err)
+		}
+
+		loopLogger := w.logger.With(
+			zap.String("scheduler", schedulerName),
+			zap.String("operation_id", op.ID),
+			zap.String("operation_definition", def.Name()),
+		)
+
+		executor, hasExecutor := w.executors[def.Name()]
+		if !hasExecutor {
+			loopLogger.Warn("operation definition has no executor")
+			w.evictOperation(ctx, loopLogger, op)
+			continue
+		}
+
+		if !def.ShouldExecute(ctx, []*operation.Operation{}) {
+			w.evictOperation(ctx, loopLogger, op)
+			continue
+		}
+
+		err = w.opManager.StartOperation(ctx, op)
+		if err != nil {
+			// NOTE: currently, we're not treating if the operation exists or
+			// not. In this case, when there is error it will be a unexpected
+			// error.
+			return fmt.Errorf("failed to start operation \"%s\" for the scheduler \"%s\"", op.ID, op.SchedulerName)
```

Comment:
We can do an improvement on the manager to recreate workers that return `false` for the `IsRunning` but in another PR. Returning error here Is important since the worker could start wasting operations, which we can also try to find a solution for.

Reply:
The only thing I would do here is to call the stop function in order to mark this worker `run` property as false.